packages/util/src/verkle.ts

```diff
@@ -322,48 +325,51 @@ export function decodeVerkleLeafBasicData(encodedBasicData: Uint8Array): VerkleL
 export function encodeVerkleLeafBasicData(account: Account): Uint8Array {
   const encodedVersion = setLengthLeft(int32ToBytes(account.version), VERKLE_VERSION_BYTES_LENGTH)
   // Per EIP-6800, bytes 1-4 are reserved for future use
-  const reservedBytes = new Uint8Array([0, 0, 0])
+  const reservedBytes = new Uint8Array([0, 0, 0, 0])
   const encodedNonce = setLengthLeft(bigIntToBytes(account.nonce), VERKLE_NONCE_BYTES_LENGTH)
   const encodedCodeSize = setLengthLeft(
     int32ToBytes(account.codeSize),
     VERKLE_CODE_SIZE_BYTES_LENGTH,
   )
   const encodedBalance = setLengthLeft(bigIntToBytes(account.balance), VERKLE_BALANCE_BYTES_LENGTH)
-  return concatBytes(encodedVersion, reservedBytes, encodedNonce, encodedCodeSize, encodedBalance)
+  return concatBytes(encodedVersion, reservedBytes, encodedCodeSize, encodedNonce, encodedBalance)
 }
 
+/**
+ * Helper method to generate the suffixes for code chunks for putting code
+ * @param numChunks number of chunks to generate suffixes for
+ * @returns number[] - an array of numbers corresponding to the code chunks being put
+ */
 export const generateChunkSuffixes = (numChunks: number) => {
-  const chunkSuffixes: number[] = new Array(numChunks)
-  for (let x = 0; x < numChunks; x++) {
-    if (x < VERKLE_CODE_OFFSET) {
-      // Suffixes for chunks 0 - 127 start at 128 and end with 255
-      chunkSuffixes[x] = x + VERKLE_CODE_OFFSET
-      continue
-    }
-    if (x < 384) {
-      // Suffixes for chunks 128 - 383 start at 0 and go to 255
-      chunkSuffixes[x] = x - VERKLE_CODE_OFFSET
-      continue
-    }
-    if (x < 640) {
-      // Suffixes for chunks 384 - 639 start at 0 and go to 255
-      chunkSuffixes[x] = x - VERKLE_CODE_OFFSET - VERKLE_NODE_WIDTH
-      continue
-    }
-    if (x > 639) {
-      // Suffixes for chunks 640 - 793 start at 0 and go to 153
-      chunkSuffixes[x] = x - VERKLE_CODE_OFFSET - 2 * VERKLE_NODE_WIDTH
-      continue
+  if (numChunks === 0) return []
+  const chunkSuffixes: number[] = new Array<number>(numChunks)
```

Comment:
```suggestion
  const chunkSuffixes = new Array<number>(numChunks)
```
This type is probably unnecessary? Should be inferred from new Array<number>(numChunks)